src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -370,7 +373,7 @@ machine(MachineType:TCC, "TCC Cache")
           // TCC will perform the atomic on the return path on Event:Data.
           // The action will invalidate the cache line if SLC is set and the address is
           // in the cache.
-          if(in_msg.isSLCSet || !WB || !presentOrAvail(in_msg.addr)) {
```

Comment:
Remind me why this is removed?

Reply:
The presentOrAvail check was intended to check if the cache line was invalid, but it wasn't quite doing that. To account for invalid, lines, I've added "Atomic" to the AtomicPassOn transition for going from I to A [here](https://github.com/gem5/gem5/pull/397/files#diff-70af2194466dfb4563c3a46e40715dfcffbf43f262eb4eac9da661c67e240996R968).